src/server/run_sched.c

```diff
@@ -253,94 +202,69 @@ find_sched_from_sock(int sock)
 	pbs_sched *psched;
 
 	for (psched = (pbs_sched*) GET_NEXT(svr_allscheds); psched; psched = (pbs_sched*) GET_NEXT(psched->sc_link)) {
-		if (psched->scheduler_sock == sock || psched->scheduler_sock2 == sock)
+		if (psched->sc_primary_conn == sock || psched->sc_secondary_conn == sock)
 			return psched;
 	}
 	return NULL;
 }
 
 /**
  * @brief
- * 		contact_sched - open connection to the scheduler and send it a command
- *		Jobid is passed if and only if the cmd is SCH_SCHEDULE_AJOB
+ * 		Connects to the Scheduler requested.
+ *
+ * @param[in]	psched	-	pointer to scheduler object
+ * @param[in] state	-	state of the scheduler
  *
- * @param[in]	cmd	- the command to send
- * @param[in]	jobid	- the jobid to send if 'cmd' is SCH_SCHEDULE_AJOB
+ * @return	void
  */
+void
+set_sched_state(pbs_sched *psched, char *state)
+{
+	set_attr_generic(&(psched->sch_attr[SCHED_ATR_sched_state]), &sched_attr_def[SCHED_ATR_sched_state], state, NULL, SET);
+	server.sv_attr[SVR_ATR_State].at_flags |= ATR_VFLAG_MODCACHE;
```

Comment:
How does the server's state change when the scheduler's state change?  There is no action function for the scheduler's state.

Reply:
I am also not sure where did that come, may copy paste error. anyways this func is removed wrt to set_attr_generic

Reply:
Well set_attr_generic() will set SET_MOD_MCACHE for the scheduler attribute since it is modifying the scheduler attribute.  I'm just unfamiliar with what the server's state changes to when a scheduler goes into scheduling.  Does the server's state go into scheduling if any scheduler moves to scheduling, or just the default scheduler?  Maybe @suresh-thelkar might know since he was involved with the multi-sched feature.

Reply:
Yes @bhroam, server's state go to scheduling if only default scheduler moves to scheduling state.